taxii2-feed/operations.py

```diff
@@ -0,0 +1,274 @@
+                return {'Content-Type': response.headers['Content-Type']}
+            if response.status_code == 200 or response.status_code == 206:
+                return response.json()
+            if self.error_msg[response.status_code]:
+                logger.error('{}'.format(response.content))
+                raise ConnectorError('{}'.format(self.error_msg[response.status_code]))
+            response.raise_for_status()
+        except requests.exceptions.SSLError as e:
+            logger.exception('{}'.format(e))
+            raise ConnectorError('{}'.format(self.error_msg['ssl_error']))
+        except requests.exceptions.ConnectionError as e:
+            logger.exception('{}'.format(e))
+            raise ConnectorError('{}'.format(self.error_msg['time_out']))
+        except Exception as e:
+            logger.exception('{}'.format(e))
+            raise ConnectorError('{}'.format(e))
+
+    def get_api_root_information(self, endpoint):
+        api_root = self.make_request(endpoint=self.server_url + endpoint, headers={'Content-Type': 'application/json'})
+        return api_root['api_roots'][0]
+
+
+def get_params(params):
+    if params.get('collection_type'):
+        params.pop('collection_type')
+    params = {k: v for k, v in params.items() if v is not None and v != ''}
+    return params
+
+
+def get_output_schema(config, params, *args, **kwargs):
+    if params.get('file_response'):
+        return ({
+            "md5": "",
+            "sha1": "",
+            "sha256": "",
+            "filename": "",
+            "content_length": "",
+            "content_type": ""
+        })
+    else:
+        return ({
+            "id": "",
+            "objects": [
+                {
+                    "created": "",
+                    "description": "",
+                    "id": "",
+                    "labels": [
+                    ],
+                    "modified": "",
+                    "name": "",
+                    "object_marking_refs": [
+                    ],
+                    "pattern": "",
+                    "type": "",
+                    "valid_from": ""
+                }
+            ],
+            "spec_version": "",
+            "type": ""
+        })
+
+
+def get_epoch(_date):
+    try:
+        pattern = '%Y-%m-%dT%H:%M:%S.%fZ' if '.' in _date else '%Y-%m-%dT%H:%M:%SZ'
+        return int(time.mktime(time.strptime(_date, pattern)))
+    except Exception as err:
+        logger.exception(str(err))
+        raise ConnectorError(str(err))
+
+
+def max_age(params, ioc):
+    if params.get("expiry") is not None and params.get("expiry") != '':
+        return get_epoch(ioc["valid_from"]) + (params.get("expiry") * 86400)
+    elif "valid_until" in ioc.keys():
+        return get_epoch(ioc["valid_until"])
+    else:
+        return None
+
+
+def tlp(TLP_AMBER, TLP_RED, TLP_WHITE, TLP_GREEN, params):
```

Comment:
why upper case for function params?

Reply:
also, is this function used? don't see its usage anywhere

Reply:
Yes, mam, That is not more used ... deleted